rust/index/src/hnsw_provider.rs

```diff
@@ -857,6 +844,11 @@ mod tests {
             .await
             .unwrap();
         let created_index_id = created_index.inner.read().hnsw_index.id;
+        provider.commit(created_index.clone()).unwrap();
```

Comment:
Modified this test to flush to S3, before testing forking on it. I'm assuming this is aligned with how we actually use these operations.